### PR TITLE
WE-768 Added MudLog GeologyIntervals

### PR DESCRIPTION
**Pull request by KindCoder-no**
## Fixes
This pull request fixes WE-768

## Description
Added MudLog GeologyIntervals to frontend and backend and fixed comments on WE-767

## Type of change
* New feature (non-breaking change which adds functionality)
* Enhancement of existing functionality

## Impacted Areas in Application

* Frontend
* Backend (API)

## Checklist:

*Communication*
* [X] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [X] Code follows the style guidelines
* [X] I have self-reviewed my code
* [X] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
 import React, { useContext, useEffect, useState } from "react";
 import NavigationContext from "../../contexts/navigationContext";
-import MudLogObject from "../../models/mudLog";
+import mudLog from "../../models/mudLog";
 import { ContentTable, ContentTableColumn, ContentTableRow, ContentType } from "./table";
 
-export interface MudLogObjectRow extends ContentTableRow {
-  message: MudLogObject;
+export interface mudLogRow extends ContentTableRow {
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx]: Interface names should start with an uppercase letter

File: Src/WitsmlExplorer.Frontend/services/mudLogService.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import mudLog, { emptymudLog } from "../models/mudLog";
+import MudLogComponent from "../models/mudLogGeologyInterval";
+import { ApiClient } from "./apiClient";
+
+export default class mudLogService {
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/services/mudLogService.tsx]: Classes should start with uppercase

File: Src/WitsmlExplorer.Frontend/models/wellbore.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import BhaRun from "./bhaRun";
 import LogObject from "./logObject";
 import Measure from "./measure";
 import MessageObject from "./messageObject";
-import MudLogObject from "./mudLog";
+import mudLog from "./mudLog";
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/models/wellbore.tsx]: All imports seem to be aliased with lowercase first letter, whilst they should be uppercase. Probably a search-replace oversight 😬

File: Src/WitsmlExplorer.Frontend/models/mudLog.tsx [hunk 4]
```diff
@@ -2,9 +2,9 @@ import ObjectOnWellbore, { emptyObjectOnWellbore } from "./objectOnWellbore";
 
 // This is disabled because more attributes will be included later
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-export default interface MudLogObject extends ObjectOnWellbore {}
+export default interface MudLog extends ObjectOnWellbore {}
 
-export function emptyMudLogObject(): MudLogObject {
+export function emptymudLog(): MudLog {
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/models/mudLog.tsx]: uppercase M på "emptymudLog"

File: Src/WitsmlExplorer.Frontend/contexts/navigationActions.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import BhaRun from "../models/bhaRun";
 import LogObject from "../models/logObject";
 import MessageObject from "../models/messageObject";
-import MudLogObject from "../models/mudLog";
+import { default as MudLog, default as mudLog } from "../models/mudLog";
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/contexts/navigationActions.tsx]: 🤔

Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/contexts/navigationActions.tsx]: I think prettier did this. (auto imports). Idk why

File: Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx (hunk 1 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx]: Fixed

File: Src/WitsmlExplorer.Frontend/services/mudLogService.tsx (hunk 2 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/services/mudLogService.tsx]: Fixed

File: Src/WitsmlExplorer.Frontend/models/wellbore.tsx (hunk 3 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/models/wellbore.tsx]: Fixed

File: Src/WitsmlExplorer.Frontend/models/mudLog.tsx (hunk 4 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/models/mudLog.tsx]: Fixed

File: Src/WitsmlExplorer.Frontend/contexts/navigationActions.tsx (hunk 5 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/contexts/navigationActions.tsx]: Fixed

**Review by janburak — APPROVED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Pull request merged**